crates/cheatnet/src/forking/state.rs

```diff
@@ -99,9 +73,9 @@ impl BlockInfoReader for ForkStateReader {
 
                 Ok(block_info)
             }
-            Ok(MaybePendingBlockWithTxHashes::PendingBlock(pending_block)) => {
-                get_pending_block_parent(self, &pending_block)
-            }
+            Ok(MaybePendingBlockWithTxHashes::PendingBlock(_)) => Err(StateReadError(
+                "Forking Pending block is not available".to_string(),
+            )),
```

Comment:
It should be `unreachable!` I think

Reply:
Probably yes 👍 didn't know about that macro

Reply:
Is it always unreachable though? We removed the tag but isn't it possible to fetch pending giving specific block number? If it is I would keep the human error.

Reply:
> Is it always unreachable though? We removed the tag but isn't it possible to fetch pending giving specific block number? If it is I would keep the human error.

It seems it's not possible to fetch the pending block with it's block number.

Reply:
@MaksymilianDemitraszek I think pending block does not have a number, only parent hash

Reply:
So I assume it is poor type encoding of JSONRpcClient?

Reply:
Nope, their type encoding is fine, `self.block_id` is of type `BlockId`. It is probably a general arg that has to be passed to methods like this (doubt there are separate methods where you pass the number explicitly)

```rust
pub enum BlockId {
    Hash(FieldElement),
    Number(u64),
    Tag(BlockTag),
}
```